cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -237,10 +277,24 @@ def from_unitary(u: np.ndarray) -> Optional['SingleQubitCliffordGate']:
         z_to = _to_pauli_transform(u @ z @ u.conj().T)
         if x_to is None or z_to is None:
             return None
-        return SingleQubitCliffordGate.from_double_map({pauli_gates.X: x_to, pauli_gates.Z: z_to})
+        return SingleQubitCliffordGate.from_cliiford_tableau(
+            _to_cliiford_tableau(x_to=x_to, z_to=z_to)
+        )
 
     def transform(self, pauli: Pauli) -> PauliTransform:
-        return self._rotation_map[pauli]
+        x_to = self._clifford_tableau.destabilizers()[0]
+        z_to = self._clifford_tableau.stabilizers()[0]
+        if pauli == pauli_gates.X:
+            to = x_to
+        elif pauli == pauli_gates.Z:
+            to = z_to
+        else:
+            to = x_to * z_to  # Y = iXZ
+            to.coefficient *= 1j
+        single_pauli_gates = (pauli_gates.X, pauli_gates.Y, pauli_gates.Z)
+        # pauli_mask returns a value between 0 and 4 for [I, X, Y, Z].
```

Comment:
where is `pauli_mask`?

Reply:
the line below: `to_gate = single_pauli_gates[to.pauli_mask[0] - 1]` in the middle of getting index.
Just realized it can also be simplified into `to_gate = Pauli._XYZ[to.pauli_mask[0] - 1]`